### PR TITLE
Improving the rutime table test

### PR DESCRIPTION
**Pull request by volhovm**
The `main_runtime_table_cfg` is failing in https://github.com/MinaProtocol/mina/pull/14725 after a minor change, so in this PR it's improved before it's used in https://github.com/MinaProtocol/mina/pull/14725.

The changes are only done to the runtime tables test (the fixed lookups test is just refactored a bit  for readability). Basically, now runtime tables are much more randomized.

### CODE REVIEW

**Comment by volhovm** [comment]
!ci-build-me

File: src/lib/pickles/test/optional_custom_gates/pickles_test_optional_custom_gates.ml [hunk 1]
```diff
@@ -281,30 +284,63 @@ let main_fixed_lookup_tables_multiple_tables_multiple_lookups () =
...
+(* nb of runtime lookup tables *)
+let runtime_lt_tables_n = 1 + Random.State.int state 10
+
+(* number of runtime lookups *)
+let runtime_lt_table_queries_n = 1 + Random.State.int state 20
+
+(* runtime lookup tables data *)
+let runtime_lt_data =
+  Array.init runtime_lt_tables_n ~f:(fun i ->
+      let table_id = i in
```
Comment (dannywillems) [inline src/lib/pickles/test/optional_custom_gates/pickles_test_optional_custom_gates.ml:296]: Could be random.

File: src/lib/pickles/test/optional_custom_gates/pickles_test_optional_custom_gates.ml [hunk 2]
```diff
@@ -281,30 +284,63 @@ let main_fixed_lookup_tables_multiple_tables_multiple_lookups () =
...
+      let table_size = 1 + Random.State.int state 100 in
+      let indices_offset = Random.State.int state 100 in
+      let first_column =
+        Array.init table_size ~f:(fun i -> indices_offset + i)
```
Comment (dannywillems) [inline src/lib/pickles/test/optional_custom_gates/pickles_test_optional_custom_gates.ml:300]: Could be more random.

**Review by dannywillems — APPROVED** [review]
LGTM. Let's have a follow-up for the more randomized tests.

**Pull request merged**